packages/react-dom/src/events/EnterLeaveEventPlugin.js

```diff
@@ -169,11 +169,14 @@ const EnterLeaveEventPlugin = {
 
     accumulateEnterLeaveDispatches(leave, enter, from, to);
 
-    if (nativeEvent === lastNativeEvent) {
-      lastNativeEvent = null;
+    if (seenNativeEvents.has(nativeEvent)) {
+      seenNativeEvents.remove(nativeEvent);
       return [leave];
     }
-    lastNativeEvent = nativeEvent;
+    if (seenNativeEvents instanceof Set) {
+      seenNativeEvents.clear();
+    }
```

Comment:
Maybe:
seenNativeEvents = typeof WeakSet === 'function' ? new WeakSet() : new Set();

Reply:
When `WeakSet` is supported then is there any harm to letting the set ‘grow’ infinitely? Something's presence in the set will be contingent on it being held on to somewhere else in your app, right?

When `WeakSet` is _not_ supported, I wonder if it's more efficient to call `clear()` rather than to construct a new `Set`.

Reply:
There's a small one. If you're leaking the thing you're holding a weak reference to, or retaining them on purpose, then this set is growing. If we only care about the last thing it's best to only track the last thing. It's unfortunate you can't simply clear a `WeakSet`.